### PR TITLE
SUS-1514: Log parsing time for GeSHi syntax highlight

### PR DESCRIPTION
**Pull request by mszabo-wikia**
https://wikia-inc.atlassian.net/browse/SUS-1514

### CODE REVIEW

File: extensions/SyntaxHighlight_GeSHi/SyntaxHighlight_GeSHi.class.php [hunk 1]
```diff
@@ -477,4 +484,19 @@ private static function tabsToSpacesLine( $line ) {
...
+	 * Log GeSHi syntax highlight parsing time with 10% sampling
+	 *
+	 * @see https://wikia-inc.atlassian.net/browse/SUS-1514
+	 * @param Title $title article title of JS page or page with a syntaxhighlight tag
+	 * @param GeSHi $geSHi
+	 */
+	private static function profileGeshiPerformance( Title $title, GeSHi $geSHi ) {
+		\Wikia\Logger\WikiaLogger::instance()->debugSampled( 0.1, 'SUS-1514 - GeSHi syntax highlight performance', [
+			'titleText' => $title->getPrefixedText(),
+			'parsingTime' => $geSHi->get_time()
```
Comment (macbre) [inline extensions/SyntaxHighlight_GeSHi/SyntaxHighlight_GeSHi.class.php]: What's the unit of the time reported here?

File: extensions/SyntaxHighlight_GeSHi/SyntaxHighlight_GeSHi.class.php [hunk 2]
```diff
@@ -477,4 +484,19 @@ private static function tabsToSpacesLine( $line ) {
...
+	 * Wikia change
+	 * Log GeSHi syntax highlight parsing time with 10% sampling
+	 *
+	 * @see https://wikia-inc.atlassian.net/browse/SUS-1514
+	 * @param Title $title article title of JS page or page with a syntaxhighlight tag
+	 * @param GeSHi $geSHi
+	 */
+	private static function profileGeshiPerformance( Title $title, GeSHi $geSHi ) {
+		\Wikia\Logger\WikiaLogger::instance()->debugSampled( 0.1, 'SUS-1514 - GeSHi syntax highlight performance', [
+			'titleText' => $title->getPrefixedText(),
```
Comment (macbre) [inline extensions/SyntaxHighlight_GeSHi/SyntaxHighlight_GeSHi.class.php]: Page URL will be reported as a part of generic logger context. Do we need to log `titleText` here?

**Review by macbre — CHANGES_REQUESTED** [review]

File: extensions/SyntaxHighlight_GeSHi/SyntaxHighlight_GeSHi.class.php (hunk 1 above)
Comment (mszabo-wikia) [reply extensions/SyntaxHighlight_GeSHi/SyntaxHighlight_GeSHi.class.php]: PHP  `microtime` format-style double generated at https://github.com/Wikia/app/blob/a2f56c08de6461f475232c399c4601eb5347d25c/extensions/SyntaxHighlight_GeSHi/geshi/geshi.php#L3677

File: extensions/SyntaxHighlight_GeSHi/SyntaxHighlight_GeSHi.class.php (hunk 2 above)
Comment (mszabo-wikia) [reply extensions/SyntaxHighlight_GeSHi/SyntaxHighlight_GeSHi.class.php]: good catch - will remove :)

**Review by macbre — APPROVED** [review]

**Pull request merged**